external/llvm-project/mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -222,7 +222,7 @@ def AMDGPU_RawBufferAtomicCmpswapOp :
       AttrSizedOperandSegments,
       AllTypesMatch<["src", "cmp", "value"]>,
       AllElementTypesMatch<["value", "memref"]>]>,
-    Arguments<(ins AnyTypeOf<[I32, I64, F32, F64]>:$src,
+    Arguments<(ins AnyType:$src,
```

Comment:
I'd call for a stricter validation here, since there's only a limited number of things you can CAS? But maybe we're punting that to the compiler

Reply:
Yes, I entertained the thought of listing out possible vector elements that are legal then I found out following code is already there and would give a much more meaningful error message that tablegen would ve given.

https://github.com/ROCm/rocMLIR/blob/3612396bca1139abf25e2ed0085fe481d275af89/external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp#L111-L135